src/app/Services/product.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core';
+import {Card, Product} from "../interfaces";
+import {BehaviorSubject, Observable} from "rxjs";
+import {products} from "../constants";
+
+@Injectable({
+    providedIn: 'root'
+})
+
+export class ProductService {
+
+    private cardListSubject = new BehaviorSubject<Card[]>([]);
+    cardList$ = this.cardListSubject.asObservable();
+
+    private productSubject = new BehaviorSubject<Product[]>([]);
```

Comment:
Tout ce qui est Observable, Subject, ou BehaviorSubject se suffix de $
```suggestion
    private _productSubject$ = new BehaviorSubject<Product[]>([]);
```